src/NumericalAlgorithms/Interpolation/Actions/SendPointsToInterpolator.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+
+namespace intrp {
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sets up points on an `InterpolationTarget` at a new `temporal_id`
+/// and sends these points to an `Interpolator`.
+///
+/// Uses:
+/// - DataBox:
+///   - `domain::Tags::Domain<3>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `Tags::IndicesOfInvalidInterpPoints`
+///   - `Tags::InterpolatedVars<InterpolationTargetTag, TemporalId>`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag>
+struct SendPointsToInterpolator {
+  template <
+      typename ParallelComponent, typename DbTags, typename Metavariables,
+      typename ArrayIndex, typename TemporalId,
+      Requires<tmpl::list_contains_v<DbTags, Tags::TemporalIds<TemporalId>>> =
+          nullptr>
+  static void apply(db::DataBox<DbTags>& box,
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const TemporalId& temporal_id) noexcept {
+    auto coords = InterpolationTarget_detail::get_block_logical_coords<
+        InterpolationTargetTag>(box, cache, temporal_id);
+    InterpolationTarget_detail::set_up_interpolation<InterpolationTargetTag>(
+        make_not_null(&box), temporal_id, coords);
+    auto& receiver_proxy =
+        Parallel::get_parallel_component<Interpolator<Metavariables>>(cache);
+    Parallel::simple_action<Actions::ReceivePoints<InterpolationTargetTag>>(
+        receiver_proxy, temporal_id, std::move(coords));
```

Comment:
Here I got confused about the overall flow: why is the `coords` data going to two places? I see it being added to `box` in `set_up_interpolation` _and_ sent to the `receiver_proxy` component. (For now, just answer in Github while I try to fix my understanding)

Reply:
Coords are not added to the box in set_up_interpolation. In set_up_interpolation, the coords are used only to determine which ones are invalid (i.e. which ones are outside of all blocks, e.g. inside an excision boundary), and for their size.  SendPointsToInterpolator is an Action called on the InterpolationTarget component. receiver_proxy is the Interpolator component not the InterpolationTarget component, which needs the coords because it needs to know what to interpolate onto.

Later, when we have interpolation without the Interpolator component, the coords will be sent to the DgElementArray (which will do the interpolation), and the InterpolationTarget will still call `set_up_interpolation` (which I hope will be completely unchanged) because it needs to know which points are valid and how many points there are, so that it knows when it has received all the points that it is expecting.

Reply:
@markscheel Thank you for clarifying. This makes sense now 👍. No changes needed in this code location. (I will leave a new comment at the code line that caused my confusion).